pandas/core/generic.py

```diff
@@ -8857,13 +8857,18 @@ def _where(
         level=None,
         errors="raise",
         try_cast=False,
+        invert=False,
+        apply_callable_other=False,
     ):
         """
         Equivalent to public method `where`, except that `other` is not
         applied as a function even if callable. Used in __setitem__.
         """
         inplace = validate_bool_kwarg(inplace, "inplace")
 
+        if apply_callable_other:
```

Comment:
why is this needed? I am *almost* always -1 on adding args to anything to fix special cases.

Reply:
This was needed because evidently when we setitem using a mask and a callable other, we don't actually want that callable to get applied (and it passes through the `_where` code path):
```python
[ins] In [1]: import pandas as pd
         ...:
         ...: inc = lambda x: x + 1
         ...:
         ...: s = pd.Series([1, 2, -1, 4])
         ...: s[s < 0] = inc
         ...: s
         ...:
Out[1]:
0                                        1
1                                        2
2    <function <lambda> at 0x7fa7fb59ef70>
3                                        4
dtype: object
[ins] In [2]: pd.__version__
Out[2]: '1.1.2'
```
But other methods like `mask` and `where` need for any functions to be evaluated to booleans. Previously that logic existed outside `_where`, but if it's moved inside then I think some control is needed over whether or not to apply the function.